contracts/ERC20Mixer.sol

```diff
@@ -26,20 +26,20 @@ contract ERC20Mixer is Mixer {
     token = _token;
   }
 
-  function _processDeposit() internal {
+  function _processDeposit() internal nonReentrant {
     require(msg.value == 0, "ETH value is supposed to be 0 for ETH mixer");
     _safeErc20TransferFrom(msg.sender, address(this), denomination);
   }
 
-  function _processWithdraw(address payable _recipient, address payable _relayer, uint256 _fee, uint256 _refund) internal {
+  function _processWithdraw(address payable _recipient, address payable _relayer, uint256 _fee, uint256 _refund) internal nonReentrant {
     require(msg.value == _refund, "Incorrect refund amount received by the contract");
 
     _safeErc20Transfer(_recipient, denomination - _fee);
     if (_fee > 0) {
       _safeErc20Transfer(_relayer, _fee);
     }
     if (_refund > 0) {
-      _recipient.transfer(_refund);
+      _recipient.call.value(_refund)("");
```

Comment:
Call status is ignored here, so refunds that were not delivered to recipients will remain on Mixer's balance forever.  Is this OK?  Other options would be to send such refunds to relayer.
Also, however this change seems to fix original issue, it also allows recipient to spend more than 2300 gas when processing incoming refund transfer.  This additional gas will be paid by relayer.  If this is not intentional, consider using “send” instead of “call”.
Also, comment about why you do not revert on failed transfer would be helpful, as people usually treat such things as possible bugs.

Reply:
https://diligence.consensys.net/blog/2019/09/stop-using-soliditys-transfer-now/

1. FYI using `transfer()` and `send()` is not safe. 
2. we will add check for status of refund tx

Reply:
Oh, I see.  Just wanted to make sure this change was intentional.